### PR TITLE
Update all Vagrant boxes to use Puppet5

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

Have all Vagrant boxes use Puppet5.

## Related Issue

This is needed for #991 Vagrantfile change to be functional.  Also should have likely been a part of #988 when Puppet 4 support was dropped.

## Motivation and Context

This module no longer tests against Puppet 4 so stop using Puppet 4 in Vagrant.

### CODE REVIEW

**Comment by treydock** [comment]
My vagrant environment is not working at the moment so been difficult to test these changes locally.

File: Vagrantfile [hunk 1]
```diff
@@ -65,20 +65,6 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     server.vm.provision :shell, :inline => "facter --custom-dir=/vagrant/lib/facter sensu_version"
   end
 
-  # This system is meant to be started without 'sensu-server' running.
-  config.vm.define "sensu-server-puppet5", autostart: false do |server|
```
Comment (ghoneycutt) [inline Vagrantfile]: This should still exist, but be puppet6

**Comment by ghoneycutt** [comment]
`amazon201703-client` fails because it is not mounting `/vagrant`. We should probably only be supporting the latest version of Amazon Linux. They do not seem to put an EOL on their versions and just encourage using the latest.

**Comment by ghoneycutt** [comment]
Both windows systems do not work. It seems like puppet is not in the PATH.

**Comment by treydock** [comment]
I've not found official amazon linux vagrant boxes so it's always been hit or miss if the community ones work.  Amazon does release docker images.

**Comment by treydock** [comment]
Windows box fixed.  Had to use x64 MSI instead of x86.  The chocolatey one won't fully work as still no chocolatey packages for sensu.

Also added commit to address #994

**Comment by ghoneycutt** [comment]
@cwjohnston Are you looking to deploy to the chocolatey repo for windows users? It would really aid in adoption by making it easier to consume.

**Comment by ghoneycutt** [comment]
@treydock let me know when the macos-client is fixed and I'll re-test everything again.

**Comment by ghoneycutt** [comment]
Released in v2.60.0

**Pull request merged**